bin/docker-image-tool.sh

```diff
@@ -0,0 +1,263 @@
+  TOTAL_JARS=$(( $TOTAL_JARS ))
+  if [ "${TOTAL_JARS}" -eq 0 ]; then
+    error "Cannot find Kyuubi JARs. This script assumes that Apache Kyuubi has first been built locally or this is a runnable distribution."
+  fi
+
+  local BUILD_ARGS=(${BUILD_PARAMS})
+
+  # If a custom Kyuubi_UID was set add it to build arguments
+  if [ -n "$KYUUBI_UID" ]; then
+    BUILD_ARGS+=(--build-arg kyuubi_uid=$KYUUBI_UID)
+  fi
+
+  local BINDING_BUILD_ARGS=(
+    ${BUILD_ARGS[@]}
+    --build-arg
+    base_img=$(image_ref kyuubi)
+  )
+
+  local BASEDOCKERFILE=${BASEDOCKERFILE:-"docker/Dockerfile"}
+  local ARCHS=${ARCHS:-"--platform linux/amd64,linux/arm64"}
+
+  (cd $(img_ctx_dir base) && docker build $NOCACHEARG "${BUILD_ARGS[@]}" \
+    -t $(image_ref kyuubi) \
+    -f "$BASEDOCKERFILE" .)
+  if [ $? -ne 0 ]; then
+    error "Failed to build Kyuubi JVM Docker image, please refer to Docker build output for details."
+  fi
+  if [ "${CROSS_BUILD}" != "false" ]; then
+  (cd $(img_ctx_dir base) && docker buildx build $ARCHS $NOCACHEARG "${BUILD_ARGS[@]}" --push \
+    -t $(image_ref kyuubi) \
+    -f "$BASEDOCKERFILE" .)
+  fi
+}
+
+function push {
+  docker_push "kyuubi"
+}
+
+function usage {
+  cat <<EOF
+Usage: $0 [options] [command]
+Builds or pushes the built-in Kyuubi Docker image.
+
+Commands:
+  build       Build image. Requires a repository address to be provided if the image will be
+              pushed to a different registry.
+  push        Push a pre-built image to a registry. Requires a repository address to be provided.
+
+Options:
+  -f                    Dockerfile to build for JVM based Jobs. By default builds the Dockerfile shipped with Kyuubi.
+  -r                    Repository address.
+  -t                    Tag to apply to the built image, or to identify the image to be pushed.
+  -n                    Build docker image with --no-cache
+  -u                    UID to use in the USER directive to set the user the main Kyuubi process runs as inside the
+                        resulting container
+  -X                    Use docker buildx to cross build. Automatically pushes.
+                        See https://docs.docker.com/buildx/working-with-buildx/ for steps to setup buildx.
+  -b                    Build arg to build or push the image. For multiple build args, this option needs to
+                        be used separately for each build arg.
+  -s                    Put the specified Spark into the Kyuubi image to be used as the internal SPARK_HOME
+                        of the container.
+
+Examples:
+  - Build image in minikube with tag "testing"
+    $0 -m -t testing build
+
+  - Build and push image with tag "v1.4.0" to docker.io/myrepo
+    $0 -r docker.io/myrepo -t v1.4.0 build
+    $0 -r docker.io/myrepo -t v1.4.0 push
+
+  - Build and push with tag "v3.0.0" and Spark-3.1.2 as base image to docker.io/myrepo
+    $0 -r docker.io/myrepo -t v3.0.0 -b BASE_IMAGE=repo/spark:3.1.2 build
```

Comment:
replace `spark` with `kyuubi`

Reply:
This `Spark-3.1.2` means base_image not tag.

Reply:
ah,  I see